### PR TITLE
chore: improve install.ps1 part 2

### PR DESCRIPTION
**Pull request by pjmagee**
- Add checksum validation
- Delete the file if the checksum fails
- Install-Dagger should also use $env:USERPROFILE
- fix for DaggerCommit regex, it should be optional (╯°□°）╯︵ ┻━┻

### CODE REVIEW

**Comment by pjmagee** [comment]
@jedevc @wingyplus Part 2️⃣

File: install.ps1 [hunk 1]
```diff
@@ -101,6 +100,15 @@ function Get-DownloadUrl {
     return "https://dl.dagger.io/dagger/releases/${DaggerVersion}/${fileName}"
 }
 
+function Get-ChecksumUrl {
+
```
Comment (jedevc) [inline install.ps1:104]: nit: newlines :eyes: (sorry, bit of a broken record)

Would be totally happy if we could have automated linting checks for this script if you like, like I'm doing in #7633 for `install.sh`.

File: install.ps1 [hunk 2]
```diff
@@ -210,6 +218,63 @@ Please check you have the right permission to do so or try to create the path ma
...
+    if ($hash.Hash -ne $Checksum) {
+        
+        Remove-Item -Path $DownloadPath
+
+        Write-Host @"
+---------------------------------------------------------------------------
+The file checksum does not match the expected checksum!
+Expected: $Checksum
+File    : $($hash.Hash)
```
Comment (jedevc) [inline install.ps1:269]: Can we add a `Got:` field as well?

File: install.ps1 (hunk 1 above)
Comment (pjmagee) [reply install.ps1:104]: no problem. Will fix this evening

File: install.ps1 [hunk 3]
```diff
@@ -210,6 +218,63 @@ Please check you have the right permission to do so or try to create the path ma
     return (Get-Item -Path $InstallPath).FullName
 }
 
+# This function extracts the checksum of the downloaded file which contains all checksums for a version
+function Get-Checksum {
+
```
Comment (wingyplus) [inline install.ps1:223]:
```suggestion
```

File: install.ps1 [hunk 4]
```diff
@@ -210,6 +218,63 @@ Please check you have the right permission to do so or try to create the path ma
...
+        [Parameter(Mandatory = $true)]
+        [string]$DownloadPath,
+        [Parameter(Mandatory = $true)]
+        [string]$Checksum
+    )
+
+    $hash = Get-FileHash -Path $DownloadPath -Algorithm SHA256
+
+    if ($hash.Hash -ne $Checksum) {
+
```
Comment (wingyplus) [inline install.ps1:262]:
```suggestion
```

File: install.ps1 (hunk 2 above)
Comment (pjmagee) [reply install.ps1:267]: Should i rename file to Got? Since `Got` would be the same as `$hash.Hash`.

Comment (jedevc) [reply install.ps1:265]: Sorry, I misread - as-is is fine.

**Review by jedevc — APPROVED** [review]

**Comment by jedevc** [comment]
@pjmagee looks like one of the commits isn't signed off, otherwise LGTM :tada:

**Comment by pjmagee** [comment]
> @pjmagee looks like one of the commits isn't signed off, otherwise LGTM 🎉

:( Sorry, I wish there was an auto-sign off, maybe there is and I need to look into it. I've basically never used that feature so I'm a bit forgettable on it.

**Comment by jedevc** [comment]
@pjmagee everything else looks good to me, so rebased and signed off the commits that needed it - hope that's alright, since the main ones in there were also signed off.

Now the github DCO check should pass, so we can merge.

**Comment by pjmagee** [comment]
> @pjmagee everything else looks good to me, so rebased and signed off the commits that needed it - hope that's alright, since the main ones in there were also signed off.
> 
> Now the github DCO check should pass, so we can merge.

Yes, I'll be sure to use -s from now on.

**Pull request merged**